### DISCUSSION TITLE
SQL syntax for GPKG layer in virtual layer

### DISCUSSION THREAD

**Discussion by Oisin-JK** [opengisch/QField#4440] [Q&A]
I'm trying to create a virtual layer in QGIS for use in QField using SQL, but cannot work out what the syntax is to refer to a layer.

My gpkg is names "survey".
My layer is named "treePt".

I cannot get a simply SQL statement to work. I have tried the following layer names in the FROM statement
"survey - treePt"
'survey - treePt'
survey - treePt
"treePt"
'treePt'
treePt

All of which throw an error "Referenced table <layer name> in query not found"

**Comment by VxTedxV** [comment]
Not sure virtual layers work in QField... Tried it today with errors at opening project in QField...

**Reply by Oisin-JK** [reply]
Virtual layers do work in QField. I use them in my current shp file based project. I'm just trying to convert to gpkg.
